test/Microsoft.Extensions.Primitives.Tests/StringSegmentTest.cs

```diff
@@ -292,6 +295,61 @@ public void StringSegment_EqualsString_Invalid()
             Assert.False(result);
         }
 
+        public static TheoryData<StringSegment, StringComparison, bool> DefaultEqualsStringSegmentData
```

Comment:
DefaultStringSegmentEqualsStringSegmentData? I want to make a distinction between type string and StringSegment.